lib/status/status.go

```diff
@@ -557,3 +572,91 @@ const (
 	// roleTag is the name of the tag containing node role
 	roleTag = "role"
 )
+
+// FromAlertManager collects alerts from the prometheus alertmanager deployed to the cluster
+func FromAlertManager(ctx context.Context, dnsAddr string) ([]*models.GettableAlert, error) {
+	client, err := httplib.GetPlanetClient(httplib.WithLocalResolver(dnsAddr))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	transport := httptransport.NewWithClient(defaults.AlertmanagerServiceAddr, "/api/v2",
+		[]string{"http"}, client)
+
+	am := alertmanager.New(transport, nil)
+	getOk, err := am.Alert.GetAlerts(alert.NewGetAlertsParams().WithContext(ctx))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return filterAlerts(getOk.Payload), nil
+}
+
+const (
+	alertname    = "alertname"
+	watchdog     = "Watchdog"
+	job          = "job"
+	satellite    = "satellite"
+	rolloutStuck = "KubeDaemonSetRolloutStuck"
+	daemonset    = "daemonset"
+	message      = "message"
+	firing       = "firing"
+	severity     = "severity"
+	critical     = "critical"
+)
+
+// filterAlerts prevents expected alerts from being returned
+func filterAlerts(alerts []*models.GettableAlert) []*models.GettableAlert {
+	filtered := []*models.GettableAlert{}
+	hasWatchdog := false
+
+	for _, alert := range alerts {
+		if alert.Labels[alertname] == watchdog {
+			hasWatchdog = true
+			// filter the Watchdog alert, which should constantly be firing
+			continue
+		}
+
+		// filter satellite alerts, because they're already collected directly from satellite
+		if job, ok := alert.Labels[job]; ok && job == satellite {
+			continue
+		}
+
+		// Prometheus detects the gravity-site election process as an error since only one pod is ever ready
+		// alertname: KubeDaemonSetRolloutStuck
+		// daemonset: gravity-site
+		// message: Only 33.33333333333333% of the desired Pods of DaemonSet kube-system/gravity-site are scheduled...
+		if name, ok := alert.Labels[alertname]; ok && name == rolloutStuck {
+			if ds, ok := alert.Labels[daemonset]; ok && ds == constants.GravityServiceName {
+				if msg, ok := alert.Annotations[message]; ok && strings.Contains(msg, "33.333333") {
+					continue
+				}
+			}
+		}
+
+		filtered = append(filtered, alert)
+	}
+
+	// if we didn't find the watchdog alert, it indicates there is a problem with the alerting system, acting as a
+	// sort of deadman switch
+	if !hasWatchdog {
+		filtered = append(filtered, &models.GettableAlert{
+			Status: &models.AlertStatus{
+				State: stringAddr(firing),
+			},
+			Annotations: models.LabelSet{
+				message: "Alertmanager watchdog failed",
+			},
+			Alert: models.Alert{
+				Labels: models.LabelSet{
+					alertname: "WatchdogDown",
+					severity:  critical,
+				},
+			},
+		})
+	}
+	return filtered
+}
+
+func stringAddr(s string) *string {
```

Comment:
This method belongs in `lib/utils/units.go` (see there are `IntPtr` and other similar methods there at the bottom).